### PR TITLE
fix(launchpad): retry create recipes if launchpad error

### PR DESCRIPTION
**Pull request by syu-w**
Fixes: #275

### CODE REVIEW

**Review by mr-cal — APPROVED** [review]

File: craft_application/launchpad/models/recipe.py [hunk 1]
```diff
@@ -244,13 +246,27 @@ def new(  # noqa: PLR0913
...
+        for attempt in range(_ATTEMPT_COUNT):
+            try:
+                snap_entry = lp.lp.snaps.new(
+                    name=name,
+                    owner=util.get_person_link(owner),
+                    store_upload=bool(store_name),
+                    auto_build=auto_build,
+                    **kwargs,
+                )
+            except lazr.restfulclient.errors.BadRequest:
```
Comment (sergiusens) [inline craft_application/launchpad/models/recipe.py:260]: emit a debug message here

File: craft_application/launchpad/models/recipe.py [hunk 2]
```diff
@@ -370,13 +386,27 @@ def new(  # noqa: PLR0913
...
-            owner=util.get_person_link(owner),
-            project=f"/{project}",
-            auto_build=auto_build,
-            **kwargs,
-        )
+        charm_entry = None
+
+        for attempt in range(_ATTEMPT_COUNT):
+            try:
+                charm_entry = lp.lp.charm_recipes.new(
```
Comment (sergiusens) [inline craft_application/launchpad/models/recipe.py:393]: emit a debug message here, something like f"Trying to Create project ({attempt}/{_ATTEMPT_COUNT}"

File: craft_application/launchpad/models/recipe.py [hunk 3]
```diff
@@ -244,13 +246,27 @@ def new(  # noqa: PLR0913
...
-            owner=util.get_person_link(owner),
-            store_upload=bool(store_name),
-            auto_build=auto_build,
-            **kwargs,
-        )
+        snap_entry = None
+
+        for attempt in range(_ATTEMPT_COUNT):
+            try:
+                snap_entry = lp.lp.snaps.new(
```
Comment (sergiusens) [inline craft_application/launchpad/models/recipe.py:253]: emit a debug message here, something like f"Trying to Create project ({attempt}/{_ATTEMPT_COUNT}"

**Review by lengau — CHANGES_REQUESTED** [review]
Thanks!

Could you add some unit tests for this, mocking out the snaps `new` method and raising exceptions an appropriate number of times to check retry-success and retry-failure cases?

**Review by lengau — APPROVED** [review]
Excellent, thanks!

**Pull request merged**